vortex-expr/src/expressions.rs

```diff
@@ -23,90 +22,94 @@ pub struct Conjunction {
     pub predicates: Vec<Predicate>,
 }
 
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Clone, PartialEq)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub enum Value {
     /// A named reference to a qualified field in a dtype.
-    Field(FieldName),
+    Field(FieldPath),
     /// A constant scalar value.
     Literal(Scalar),
 }
 
+#[derive(Clone, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Predicate {
+    pub left: FieldPath,
+    pub op: Operator,
+    pub right: Value,
+}
+
+pub fn lit<T: Into<Scalar>>(n: T) -> Value {
+    Value::Literal(n.into())
+}
+
 impl Value {
-    pub fn field(field_name: impl Into<FieldName>) -> Value {
-        Field(field_name.into())
-    }
     // comparisons
-    pub fn eq(self, other: Value) -> Predicate {
+    pub fn eq(self, field: impl Into<FieldPath>) -> Predicate {
         Predicate {
-            left: self,
+            left: field.into(),
             op: Operator::EqualTo,
-            right: other,
+            right: self,
         }
     }
 
-    pub fn not_eq(self, other: Value) -> Predicate {
+    pub fn not_eq(self, field: impl Into<FieldPath>) -> Predicate {
         Predicate {
-            left: self,
+            left: field.into(),
             op: Operator::NotEqualTo,
-            right: other,
+            right: self,
         }
     }
 
-    pub fn gt(self, other: Value) -> Predicate {
+    pub fn gt(self, field: impl Into<FieldPath>) -> Predicate {
```

Comment:
While Eq is reflexive the others are not so I think you need to adjust the ops to be opposite or have things derive out of FieldPath which I think is more intuitive